datalad/support/gitrepo.py

```diff
@@ -3933,6 +3991,10 @@ def save_(self, message=None, paths=None, _status=None, **kwargs):
                         str(cand_sm.relative_to(self.pathobj)),
                         url=None,
                         name=None,
+                        # MIH: this looks strange, why would we want to check
+                        # out a branch that matches the name of a branch
+                        # in the parentds, we don't even know if there is
+                        # such a branch...
```

Comment:
You'll have to ask yourself that: 2fda83e2ca (ENH: Record the active branch of a subdataset in the parent, 2019-10-20).  Based on that subject and the fact that `self.get_active_branch()` is evaluated for each `cand_sm`, I'd guess you were intending to specify the currently checked out branch in `cand_sm`, not the parent.